tests/test_simple.py

```diff
@@ -374,3 +374,113 @@ def test_older_minor(self, parser, monkeypatch):
 
     # No test for older major versions as that case is currently impossible with
     # 1.0 as the only possible version.
+
+
+class TestIndexing:
```

Comment:
I felt this was appropriate, to make a test class specific to the indexing functionality.

I am missing one type of test: whitebox-testing for the indexter functionality. That will come after some of the discussions are resolved (in particular, should I have employed the `sdist` detection at this early stage?).